flexinput/src/main/java/com/lytefast/flexinput/fragment/AddContentDialogFragment.java

```diff
@@ -239,32 +239,59 @@ public void launchFileChooser() {
         .setType("image/*")
         .putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
 
-    final String fileBrowserAction = (Build.VERSION.SDK_INT < Build.VERSION_CODES.KITKAT)
-        ? Intent.ACTION_GET_CONTENT : Intent.ACTION_OPEN_DOCUMENT;
-    final Intent sysBrowserIntent =
-        new Intent(fileBrowserAction)
-            .setType("*/*")
-            .addCategory(Intent.CATEGORY_OPENABLE)
-            .addCategory(Intent.CATEGORY_DEFAULT)
-            .putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true);
+    final List<Intent> allIntents = getAllIntents();
+    final Parcelable[] intentArray = new Parcelable[allIntents.size()];
+    allIntents.toArray(intentArray);
 
-    Intent chooserIntent = Intent.createChooser(sysBrowserIntent, getLauncherString())
-        .putExtra(Intent.EXTRA_INITIAL_INTENTS, new Parcelable[] {imagePickerIntent, getGoogleDriveIntent(), });
+    Intent chooserIntent = Intent.createChooser(imagePickerIntent, getLauncherString())
+        .putExtra(Intent.EXTRA_INITIAL_INTENTS, intentArray);
     startActivityForResult(chooserIntent, REQUEST_FILES);
   }
 
+  protected List<Intent> getAllIntents() {
+    final String[] mimetypes = {"text/*", "image/*", "video/*"};
+    List<ResolveInfo> resolveInfos = getContext().getPackageManager()
+        .queryIntentActivities(
+            new Intent(Intent.ACTION_GET_CONTENT)
+                .setType("application/*")
+                .putExtra(Intent.EXTRA_MIME_TYPES, mimetypes)
```

Comment:
I've read that using `EXTRA_MIME_TYPES` will not work on Android < 19 API, has this been tested?

https://issuetracker.google.com/issues/36987231